exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestNestedLoopJoin.java

```diff
@@ -333,4 +342,96 @@ public void testNlJoinWithLargeRightInputSuccess() throws Exception {
       test(RESET_JOIN_OPTIMIZATION);
     }
   }
+
+  private void buildFile(String fileName, String[] data, File testDir) throws IOException {
+    try(PrintWriter out = new PrintWriter(new FileWriter(new File(testDir, fileName)))) {
+      for (String line : data) {
+        out.println(line);
+      }
+    }
+  }
+
+  @Test
+  public void testNestedLeftJoinWithEmptyTable() throws Exception {
+    test(DISABLE_JOIN_OPTIMIZATION);
+    test(DISABLE_NLJ_SCALAR);
+    test(DISABLE_HJ);
+    test(DISABLE_MJ);
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher);
+    File testDir = null;
+    try {
+      ClusterFixture cluster = builder.build();
+      ClientFixture client = cluster.clientFixture();
+      testDir = dirTestWatcher.getTmpDir();
+      cluster.defineWorkspace("dfs", "data", testDir.getAbsolutePath(), "json");
+      buildFile("dept.json", new String[0], testDir);
+      QueryBuilder query = client.queryBuilder().sql("select * from cp.`employee.json` emp left outer join dfs.data.`dept.json` as dept on dept.manager = emp.`last_name`");
+      assert(query.futureSummary().get().recordCount() == 1155);
+    } catch (RuntimeException ex) {
+      throw ex;
+    } finally {
+      if (testDir != null) {
+        testDir.delete();
+      }
+      test(RESET_JOIN_OPTIMIZATION);
+      test(ENABLE_NLJ_SCALAR);
+      test(ENABLE_HJ);
+      test(ENABLE_MJ);
```

Comment:
The above probably assumes the value for each option. This has caused problems, especially during failures. Suggest to instead use `resetSessionOption("option.name")`